compliance_checker/cf/util.py

```diff
@@ -930,16 +930,35 @@ def get_flag_variables(nc):
     return flag_variables
 
 
+def extract_grid_mapping_names(grid_mapping_string):
+    """
+    Extracts all grid mapping variable names from a grid_mapping string.
+
+    :param str grid_mapping_string: The grid_mapping attribute string
+    :return list[str]: List of grid mapping variable names
+    """
+    return (
+        re.findall(r"\b(\w+):", grid_mapping_string)
+        if ":" in grid_mapping_string
+        else grid_mapping_string.split()
+    )
+
+
```

Comment:
changed it to include the only two possible formats:

Key-value-like: "key: value"  (key is the grid mapping variables
Space-separated names: "x y z"  ( x y z are the grid mapping variables)